lib/mapbox-gl-geocoder.css

```diff
@@ -3,137 +3,174 @@
-    .mapboxgl-ctrl-geocoder ul > li.active > a {
-      color:#202020;
-      background-color:#e3e3e3;
-      text-decoration:none;
-      cursor:pointer;
-      }
-
-@-webkit-keyframes rotate { from { -webkit-transform: rotate(0deg); } to { -webkit-transform: rotate(360deg); } }
-   @-moz-keyframes rotate { from { -moz-transform: rotate(0deg); } to { -moz-transform: rotate(360deg); } }
-    @-ms-keyframes rotate { from { -ms-transform: rotate(0deg); } to { -ms-transform: rotate(360deg); } }
-        @keyframes rotate { from { transform: rotate(0deg); } to { transform: rotate(360deg); } }
-
-/* icons */
+  background-color: #fff;
+  border-radius: 4px;
+  left: 0;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+  position: absolute;
+  width: 100%;
+  top: 110%; /* fallback */
+  top: calc(100% + 6px);
+  z-index: 1000;
+  overflow: hidden;
+  font-size: 13px;
+}
+
+.mapboxgl-ctrl-bottom-left .mapboxgl-ctrl-geocoder ul,
+.mapboxgl-ctrl-bottom-right .mapboxgl-ctrl-geocoder ul {
+  top: auto;
+  bottom: 100%;
+}
+
+.mapboxgl-ctrl-geocoder ul > li > a {
+  cursor: default;
+  display: block;
+  padding: 6px 12px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  color: #404040;
+}
+
+.mapboxgl-ctrl-geocoder ul > li > a:hover {
+  color: #404040;
+  background-color: #f3f3f3;
+  text-decoration: none;
+  cursor: pointer;
+}
+
+/* Icons */
 .geocoder-icon {
-  display:inline-block;
-  width:20px;
-  height:20px;
-  vertical-align:middle;
-  speak:none;
-  background-repeat:no-repeat;
+  display: inline-block;
+  width: 20px;
+  height: 20px;
+  vertical-align: middle;
+  speak: none;
+  background-repeat: no-repeat;
+  background-position: center;
+}
+
+.mapboxgl-ctrl-geocoder .geocoder-icon-search {
+  position: absolute;
+  top: 10px;
+  left: 9px;
+}
+
+.mapboxgl-ctrl-geocoder .geocoder-icon-close {
+  padding: 0;
+  margin: 0;
+  background-color: #757575;
+  border: none;
+  border-radius: 100%;
+  cursor: pointer;
+  width: 16px;
+  height: 16px;
+  margin-top: 2px;
+}
+
+.mapboxgl-ctrl-geocoder .geocoder-icon-close:hover {
+  background-color: #636363;
+}
+
+.geocoder-icon-search {
+  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='18' height='18'%3E%3Cpath d='M7.4 2.5c-2.7 0-4.9 2.2-4.9 4.9s2.2 4.9 4.9 4.9c1 0 1.8-.2 2.5-.8l3.7 3.7c.2.2.4.3.8.3.7 0 1.1-.4 1.1-1.1 0-.3-.1-.5-.3-.8L11.4 10c.4-.8.8-1.6.8-2.5.1-2.8-2.1-5-4.8-5zm0 1.6c1.8 0 3.2 1.4 3.2 3.2s-1.4 3.2-3.2 3.2-3.3-1.3-3.3-3.1 1.4-3.3 3.3-3.3z' fill='%23757575'/%3E%3C/svg%3E");
+}
+
+.geocoder-icon-close {
+  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='8' height='8'%3E%3Cpath d='M.8 0C.4 0 0 .4 0 .8c0 .2.1.4.3.5L2.9 4 .3 6.6c-.2.2-.3.4-.3.6 0 .4.4.8.8.8.2 0 .4-.1.6-.3L4 5.1l2.6 2.6c.1.2.4.2.6.2.4 0 .8-.4.8-.8 0-.2-.1-.4-.2-.6L5.1 4l2.6-2.7c.2-.1.3-.3.3-.5 0-.4-.4-.8-.8-.8-.2 0-.4.1-.6.2L4 2.8 1.4.2C1.2.1 1 0 .8 0z' fill='%23fff'/%3E%3C/svg%3E");
+  display: none;
+}
```

Comment:
Could we implement the svg icons in such a way that their properties are style-able with CSS (for example, by inlining the CSS)?